From ece0834342e1dc1c6f18b5cf226e6e15644e07c4 Mon Sep 17 00:00:00 2001
From: Jan Wyszynski <jan.wyszynski@joyent.com>
Date: Fri, 5 Jan 2018 00:58:33 +0000
Subject: [PATCH] MANATEE-389 deposed array is missing in stateBackfill
 MANATEE-390 check for deposed array's existence in manatee-adm history

---
 lib/adm.js | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/lib/adm.js b/lib/adm.js
index 6e642b0..411ef5d 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -1259,6 +1259,9 @@ function stateBackfill(opts, cb) {
             if (!stat.async) {
                 stat.async = [];
             }
+            if (!stat.deposed) {
+                stat.deposed = [];
+            }
             stat.generation = 0;
             stat.initWal = '0/0000000';
             stat.freeze = {
@@ -1967,8 +1970,8 @@ function annotateHistoryNode(evt, last)
 
     newpeers = {};
     oldpeers = {};
-    nst.deposed.forEach(function (a) { newpeers[a.zoneId] = true; });
-    lst.deposed.forEach(function (a) { oldpeers[a.zoneId] = true; });
+    (nst.deposed || []).forEach(function (a) { newpeers[a.zoneId] = true; });
+    (lst.deposed || []).forEach(function (a) { oldpeers[a.zoneId] = true; });
     for (h in newpeers) {
         if (!oldpeers.hasOwnProperty(h))
             changes.push(sprintf('"%s" deposed',
-- 
2.21.0

